src/layers/decoder_layer.h

```diff
@@ -55,35 +55,69 @@ class Decoder {
 
     int getLayerId() { return layerIdx; }
 
-    void setWeights(DecoderContext *ctx, std::vector<float *> &params, bool trans = true) {
-        const float *queryWeight = params[0];
-        const float *queryBias = params[1];
-        const float *keyWeight = params[2];
-        const float *keyBias = params[3];
-        const float *valueWeight = params[4];
-        const float *valueBias = params[5];
-        const float *attnOutWeight = params[6];
-        const float *attnOutBias = params[7];
-        const float *gamma1 = params[8];
-        const float *beta1 = params[9];
-
-        attn.setWeights(ctx, queryWeight, queryBias, keyWeight, keyBias, valueWeight, valueBias, attnOutWeight,
-                attnOutBias, gamma1, beta1, trans);
-
-        std::vector<float *> mlpParams(params.begin() + 10, params.end());
-        mlp.setWeights(ctx, mlpParams, trans);
+    void setWeights(DecoderContext *ctx, std::vector<void *> &params, bool trans = true, int type = 0) {
```

Comment:
The type could use xft:data_type: https://github.com/intel/xFasterTransformer/blob/main/include/dtype.h#L18